src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -159,17 +159,23 @@ class SaveMode(Enum):
     TRUNCATE = "truncate"
 
 
+class TableCreationSource(Enum):
+    EXPLICIT_USER_COMMAND = "explicit_user_command"
```

Comment:
that doesn't really mean EXPLICIT_USER_COMMAND, right? sometimes, we may have our internal API call that may does this also, for example, to_snowpark_pandas. Maybe w can just say Others, and add a comment here that it will be coming from other save_as_table call